src/Core/Context.idr

```diff
@@ -649,22 +650,11 @@ data Transform : Type where
 ||| during codegen.
 public export
 data BuiltinType : Type where
-    ||| A built-in 'Nat'-like type
-    ||| 'NatLike : [index ->] Type'
-    ||| 'SLike : {0 _ : index} -> NatLike [index] -> NatLike [f index]'
-    ||| 'ZLike : {0 _ : index} -> NatLike [index]'
-    BuiltinNatural : BuiltinType
-    -- All the following aren't implemented yet
-    -- but are here to reduce number of TTC version changes
-    NaturalPlus : BuiltinType
-    NaturalMult : BuiltinType
-    NaturalToInteger : BuiltinType
-    IntegerToNatural : BuiltinType
+    MkBuiltinType : String -> BuiltinType
```

Comment:
Can you comment on this change? We're moving from an enumerated type that can be checked
for exhaustivity / may raise warnings if a pattern is mistyped to something stringly-typed.
This seems like a net loss.

Reply:
My thinking was fewer ttc version changes because they are really annoying, especially when you're working with multiple compilers with different versions, but I can change it back if that would be safer/better. Although I reckon we can stick with just BuiltinNatural, NaturalToInteger and IntegerToNatural and define the rest in terms of those, so that wouldn't actually be too many ttc version changes now I think about it.

Reply:
It is a bit annoying when rebuilding the compiler but I think it's worth it.